app/connectors/GovernmentGatewayConnector.scala

```diff
@@ -17,22 +17,25 @@
 package connectors
 
 import audit.EnrolEvent
-import config.{AMLSAuditConnector, ApplicationConfig, WSHttp}
+import config.{AMLSAuditConnector, AppConfig, WSHttp}
 import exceptions.{DuplicateEnrolmentException, InvalidEnrolmentCredentialsException}
-import models.governmentgateway.{EnrolmentRequest}
+import javax.inject.Inject
+import models.governmentgateway.EnrolmentRequest
 import play.api.Logger.{debug, warn}
 import play.api.libs.json.{Json, Writes}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
 import uk.gov.hmrc.play.audit.model.Audit
+import utils.AuditHelper
 
 import scala.concurrent.{ExecutionContext, Future}
-import uk.gov.hmrc.http.{CorePost, HeaderCarrier, HttpResponse}
-import utils.AuditHelper
 
-trait GovernmentGatewayConnector {
+class GovernmentGatewayConnector @Inject()(val http: WSHttp,
+                                           val appConfig: AppConfig) {
 
-  protected[connectors] def http: CorePost
-  protected def enrolUrl: String
-  private[connectors] def audit: Audit
+  val audit: Audit = new Audit(AuditHelper.appName, AMLSAuditConnector)
+  protected def enrolUrl = appConfig.enrolUrl
+  private[connectors] val duplicateEnrolmentMessage = "The service HMRC-MLR-ORG requires unique identifiers"
```

Comment:
Should these strings be in the 'Messages' file or do we not have access to that here?

Reply:
looks like messages api doesn't like to work in this case, so I'd rather stay with current solution if you don't mind